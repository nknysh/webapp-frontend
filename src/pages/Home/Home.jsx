import React from 'react';
import { Helmet } from 'react-helmet';

import { Hero, Markdown } from 'components';
import { ErrorBoundary } from 'hoc/ErrorBoundary';

import heroData from 'config/data/home/hero.md';
import tempHeroImage from './assets/temp-hero-image.jpg';

import { propTypes } from './Home.props';
import { StyledHome } from './Home.styles';

export const Home = () => {
  return (
    <StyledHome>
      <Helmet>
        <title>Pure Escapes</title>
      </Helmet>
      <Hero full image={tempHeroImage}>
        <Markdown>{heroData}</Markdown>
      </Hero>
    </StyledHome>
  );
};

Home.propTypes = propTypes;

export default ErrorBoundary(Home);
