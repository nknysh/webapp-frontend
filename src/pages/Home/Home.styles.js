import styled, { css } from 'styled-components';
import { Icon } from '@material-ui/core';
import { Parallax } from 'react-parallax';

import { theme, h2Styling, h1Styling } from 'styles';

export const StyledHome = styled.div`
  height: 100%;
  width: 100%;
`;

export const MoveTo = styled(Icon)`
  position: absolute;
  color: ${theme.colors.white};
  text-align: center;
  top: -100px;
  z-index: 30;
  width: 100% !important;
  font-size: 62px !important;
  cursor: pointer;
`;

export const HomeSection = styled.div`
  position: relative;

  ${({ ['data-striped']: striped }) =>
    striped &&
    css`
      background-color: ${theme.colors['light-blue']};
    `}
  }
`;

export const HomeHero = styled(Parallax)``;

export const HeroShim = styled.div`
  height: 50vh;
`;

export const HomeContainer = styled.div`
  text-align: center;
  padding: ${theme.spacing.gutter * 5}px;

  ${props => props.theme.breakpoints.tablet`
    padding: 120px ${theme.spacing.gutter}px;
    max-width: 600px;
    margin: 0 auto;
  `}

  h1, h2 {
    ${h1Styling}
  }

  h2 {
    ${h2Styling}
  }

  p,
  .small {
    color: ${theme.palette.primary};
  }

  .small {
    font-size: ${theme.fonts.sizes.normal}px;
  }
`;
