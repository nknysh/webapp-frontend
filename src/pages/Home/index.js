import { lazy } from 'react';

export { default } from './Home';

export const AsyncHome = lazy(() => import('./Home'));
