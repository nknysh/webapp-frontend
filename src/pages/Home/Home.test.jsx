import React from 'react';

import { Home } from './Home';

const getComponent = props => shallow(<Home {...props} />);

describe('<Home />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      expect(getComponent()).toMatchSnapshot();
    });
  });
});
