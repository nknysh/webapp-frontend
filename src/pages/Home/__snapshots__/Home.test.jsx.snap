// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render matches snapshot 1`] = `
<Homestyles__StyledHome>
  <Hero
    full={true}
    media={
      Object {
        "image": "",
        "video": Object {
          "path": "",
          "type": "mp4",
        },
      }
    }
    offsetBy={0}
  >
    <Markdown />
  </Hero>
  <Homestyles__HomeSection>
    <Homestyles__MoveTo
      onClick={[Function]}
      onTouchMove={[Function]}
    >
      keyboard_arrow_down
    </Homestyles__MoveTo>
    <Homestyles__HomeContainer>
      <Markdown />
    </Homestyles__HomeContainer>
  </Homestyles__HomeSection>
  <Homestyles__HomeSection
    data-striped={true}
  >
    <Homestyles__HomeContainer>
      <Markdown />
    </Homestyles__HomeContainer>
  </Homestyles__HomeSection>
  <Homestyles__HomeSection>
    <Homestyles__HomeContainer>
      <Markdown />
    </Homestyles__HomeContainer>
  </Homestyles__HomeSection>
  <Homestyles__HomeSection>
    <Homestyles__HomeContainer>
      <Markdown />
    </Homestyles__HomeContainer>
  </Homestyles__HomeSection>
  <Homestyles__HomeSection>
    <Homestyles__HomeContainer>
      <Markdown />
    </Homestyles__HomeContainer>
  </Homestyles__HomeSection>
</Homestyles__StyledHome>
`;
